### PR TITLE
Support hybrid targets for 'webkit'.

### PR DESCRIPTION
**Pull request by jacobbramley**
This is a prerequisite for some CapableVMs work (https://github.com/capablevms/webkit/issues/11).

### CODE REVIEW

File: pycheribuild/projects/cross/webkit.py [hunk 1]
```diff
@@ -29,7 +30,7 @@
 #
 
 from enum import Enum
-
+from ...config.compilation_targets import CompilationTargets
```
Comment (jrtc27) [inline pycheribuild/projects/cross/webkit.py:33]: Can be imported from crosscompileproject

File: pycheribuild/projects/cross/qt5.py [hunk 2]
```diff
@@ -753,6 +755,10 @@ class BuildICU4C(CrossCompileAutotoolsProject):
     native_install_dir = DefaultInstallDir.CHERI_SDK
     make_kind = MakeCommandKind.GnuMake
     needs_native_build_for_crosscompile = True
+    supported_architectures = list(
```
Comment (jrtc27) [inline pycheribuild/projects/cross/qt5.py:758]: I think we'd normally spell this `supported_architectures = CompilationTargets.ALL_CHERIBSD_TARGETS_WITH_HYBRID + [CompilationTargets.NATIVE]`

File: pycheribuild/projects/cross/qt5.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 # Copyright (c) 2017 Alex Richardson
 # All rights reserved.
 #
+# Copyright (c) 2022 Arm Limited and/or its affiliates <open-source-office@arm.com>
```
Comment (jrtc27) [inline pycheribuild/projects/cross/qt5.py:5]: Is this change really copyrightable (especially once every part of it is rewritten by me...)?

Comment (jacobbramley) [reply pycheribuild/projects/cross/qt5.py:5]: I am not a lawyer, but my understanding is broadly that copyrightability depends on jurisdiction, and presumably many other things. In any case, Arm asks us (by default) to update copyright statements on _all_ contributions so we don't need to apply our own personal heuristics, which will inevitably be wrong somewhere.

> especially once every part of it is rewritten by me
Well that makes things much easier, yes :-)

**Comment by arichardson** [comment]
Would passing "--enable-hybrid-for-purecap-rootfs-targets" work for you or exporting `CHERIBUILD_ENABLE_HYBRID_FOR_PURECAP_ROOTFS_TARGETS=1` in the environment?

File: pycheribuild/projects/cross/qt5.py [hunk 4]
```diff
@@ -753,6 +755,10 @@ class BuildICU4C(CrossCompileAutotoolsProject):
     native_install_dir = DefaultInstallDir.CHERI_SDK
     make_kind = MakeCommandKind.GnuMake
     needs_native_build_for_crosscompile = True
+    supported_architectures = list(
+        set(CompilationTargets.ALL_SUPPORTED_CHERIBSD_AND_HOST_TARGETS) |
```
Comment (arichardson) [inline pycheribuild/projects/cross/qt5.py:759]: Are you sure you intend to build against a hybrid rootfs and not the lib64 libraries in a purecap rootfs? In the latter case I'd suggest `    supported_architectures = CompilationTargets.ALL_CHERIBSD_TARGETS_WITH_HYBRID_FOR_PURECAP_ROOTFS + [
        CompilationTargets.NATIVE]`

**Review by arichardson — CHANGES_REQUESTED** [review]

File: pycheribuild/projects/cross/qt5.py (hunk 4 above)
Comment (jacobbramley) [reply pycheribuild/projects/cross/qt5.py:759]: We do actually run hybrid tests on a hybrid rootfs.

Comment (arichardson) [reply pycheribuild/projects/cross/qt5.py:759]: Is there a specific reason you are using a hybrid rootfs or could you change over to a purecap one?

Comment (jacobbramley) [reply pycheribuild/projects/cross/qt5.py:759]: The original reason was simply that it seemed like the obvious, simplest thing to do. However, it does seem that we can run our hybrid builds perfectly well with `morello-webkit-morello-hybrid-for-purecap-rootfs`.

Given that, and also that @jrtc27 wanted to rewrite it differently anyway, I propose just closing this PR.

**Pull request closed without merging**